codec/dagjson/roundtrip_test.go

```diff
@@ -0,0 +1,80 @@
+package dagjson
+
+import (
+	"bytes"
+	"testing"
+
+	. "github.com/warpfork/go-wish"
+
+	"github.com/ipld/go-ipld-prime/fluent"
+	basicnode "github.com/ipld/go-ipld-prime/node/basic"
+)
+
+var n = fluent.MustBuildMap(basicnode.Style__Map{}, 4, func(na fluent.MapAssembler) {
+	na.AssembleDirectly("plain").AssignString("olde string")
+	na.AssembleDirectly("map").CreateMap(2, func(na fluent.MapAssembler) {
+		na.AssembleDirectly("one").AssignInt(1)
```

Comment:
What about, for the common `string` key case, instead of a `MapAssembler`, this closure could be given a function as an argument to use directly to set the key, something like:

```go
...CreateMapDirectly(2, func(Assign func (string) (NodeAssembler)) {
  Assign("one").AssignInt(1)
  Assign("two").AssignInt(2)
})
```

There's still a need for a full `NodeAssembler`

Reply:
Possible, but I had to read that about three times before I figured out what it was saying, and I think adding more nested parens to the closure's args types is an _extremely_ bad idea since people have to (unfortunately) keep re-typing those all the time, so overall I don't think I like it.